pkg/publicapi/validator.go

```diff
@@ -21,12 +19,12 @@ type CustomValidator struct {
 
 func (cv *CustomValidator) Validate(i interface{}) error {
 	if err := cv.validator.Struct(i); err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+		return bad.Input(err)
 	}
 	if v, ok := i.(validatable); ok {
 		err := v.Validate()
 		if err != nil {
-			return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+			return bad.Input(err)
```

Comment:
From a readability perspective this is a step backwards, how do I know/specify which status code is being returned for other types of errors in the API?

Reply:
I would contend that it's a step forwards – what status code to return is a concern only of the API. The idea is to define metadata on the error so that the API is able to make such a distinction in a single place, rather than having to wrap every error with a status code and have that logic spread out thinly across the whole codebase.